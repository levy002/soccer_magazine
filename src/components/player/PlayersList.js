import React, { useEffect, useMemo, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPlayers } from '../../redux/player/Player';
import Player from './Player';
import logo from '../../image/logo.jpeg';
import PlayerCategory from '../playerCategory/PlayerCategory';

const PlayersList = () => {
  const players = useSelector((state) => state.players);
  const [category, setCategegory] = useState('All');
  const dispatch = useDispatch();

  useEffect(() => {
    if (players.length === 0) {
      dispatch(getPlayers());
    }
  }, []);

  const handleCategory = (e) => {
    setCategegory(e);
  };

  const getfilteredList = () => {
    if (category === 'All') {
      return players;
    }
    return players.filter((player) => player.position_name === category);
  };

  const filteredList = useMemo(getfilteredList, [category, players]);

  return (
    <>
      <div>
        <img src={logo} alt="Logo" />
      </div>
      <PlayerCategory handleCategory={handleCategory} />
      <ul>
        {filteredList.map((p) => (
          <li key={p.id}>
            <Player
              photo={p.photo}
              name={p.name}
              id={p.id}
            />
          </li>
        ))}
      </ul>
    </>
  );
};

export default PlayersList;
