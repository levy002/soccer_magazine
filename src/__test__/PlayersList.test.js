import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import PlayersList from '../components/player/PlayersList';
import store from '../redux/configureStore';

describe('test PlayersList component', () => {
  test('check whether the component renders correctly', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <PlayersList />
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
