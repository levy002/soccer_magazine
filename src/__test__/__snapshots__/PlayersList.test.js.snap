// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PlayersList component check whether the component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex flex-col bg-header-background bg-cover h-52"
      >
        <div
          class="flex justify-center items-center gap-2 h-full backdrop-blur-sm"
        >
          <img
            alt="Logo"
            class="rounded-full w-16 h-16"
            src="logo.jpeg"
          />
          <h2
            class="text-green-500 font-bold text-3xl"
          >
            Soccer
             
            <br />
             
            Magazine
          </h2>
        </div>
      </header>
      <div
        class="flex items-center justify-center gap-3 p-1"
      >
        <p
          class="text-purple-700"
        >
          Select Category:
        </p>
        <select
          class="p-2 px-3 rounded text-purple-700 bg-gray-100"
        >
          <option
            class="text-purple-700"
            value="All"
          >
            All
          </option>
          <option
            class="text-purple-700"
            value="Defender"
          >
            Defender
          </option>
          <option
            class="text-purple-700"
            value="Midfielder"
          >
            Midfielder
          </option>
          <option
            class="text-purple-700"
            value="Goalkeeper"
          >
            Goalkeeper
          </option>
          <option
            class="text-purple-700"
            value="Forward"
          >
            Forward
          </option>
        </select>
      </div>
      <ul
        class="flex flex-wrap flex-1 justify-around bg-gray-100 py-3"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="flex flex-col bg-header-background bg-cover h-52"
    >
      <div
        class="flex justify-center items-center gap-2 h-full backdrop-blur-sm"
      >
        <img
          alt="Logo"
          class="rounded-full w-16 h-16"
          src="logo.jpeg"
        />
        <h2
          class="text-green-500 font-bold text-3xl"
        >
          Soccer
           
          <br />
           
          Magazine
        </h2>
      </div>
    </header>
    <div
      class="flex items-center justify-center gap-3 p-1"
    >
      <p
        class="text-purple-700"
      >
        Select Category:
      </p>
      <select
        class="p-2 px-3 rounded text-purple-700 bg-gray-100"
      >
        <option
          class="text-purple-700"
          value="All"
        >
          All
        </option>
        <option
          class="text-purple-700"
          value="Defender"
        >
          Defender
        </option>
        <option
          class="text-purple-700"
          value="Midfielder"
        >
          Midfielder
        </option>
        <option
          class="text-purple-700"
          value="Goalkeeper"
        >
          Goalkeeper
        </option>
        <option
          class="text-purple-700"
          value="Forward"
        >
          Forward
        </option>
      </select>
    </div>
    <ul
      class="flex flex-wrap flex-1 justify-around bg-gray-100 py-3"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
